File: backend/src/main/java/az/project/VocabularyBuilder/vocabulary/dto/FilterVocabularyEntryForm.java
package az.project.VocabularyBuilder.vocabulary.dto;

import az.project.VocabularyBuilder.vocabulary.domain.ContextSource;
import az.project.VocabularyBuilder.vocabulary.domain.MasteryStatus;
import az.project.VocabularyBuilder.vocabulary.domain.PartOfSpeech;

public record FilterVocabularyEntryForm(String word, MasteryStatus masteryStatus, PartOfSpeech partOfSpeech,
                                        ContextSource contextSource) {
}

File: frontend/src/app/features/vocabulary-quiz/models/filterVocabularyEntryForm.ts
import { ContextSource } from "../../../shared/enums/contextSource";
import { MasteryStatus } from "../../../shared/enums/masterStatus";
import { PartOfSpeech } from "../../../shared/enums/partOfSpeech";

export interface FilterVocabularyEntryForm {
  word: string;
  masteryStatus: MasteryStatus | null;
  partOfSpeech: PartOfSpeech | null;
  contextSource: ContextSource | null;
}